aws/logs_monitoring/enhanced_lambda_metrics.py

```diff
@@ -303,6 +324,84 @@ def get_forwarder_telemetry_prefix_and_tags():
     return DD_FORWARDER_TELEMETRY_NAMESPACE_PREFIX, DD_FORWARDER_TELEMETRY_TAGS
 
 
+def send_forwarder_internal_metrics(name, additional_tags=[]):
+    """Send forwarder's internal metrics to DD"""
+    prefix, tags = get_forwarder_telemetry_prefix_and_tags()
+    lambda_stats.distribution(
+        "{}.{}".format(prefix, name),
+        1,
+        tags=tags + additional_tags,
+    )
+
+
+def get_last_modified_time(s3_file):
+    last_modified_str = s3_file["ResponseMetadata"]["HTTPHeaders"]["last-modified"]
+    last_modified_date = datetime.datetime.strptime(
+        last_modified_str, "%a, %d %b %Y %H:%M:%S %Z"
+    )
+    last_modified_unix_time = int(last_modified_date.strftime("%s"))
+    return last_modified_unix_time
+
+
+def acquire_s3_cache_lock():
+    """Acquire cache lock"""
+    cache_lock_object = s3_client.Object(DD_S3_BUCKET_NAME, DD_S3_CACHE_LOCK_FILENAME)
+    try:
+        file_content = cache_lock_object.get()
+
+        # check lock file expiration
+        last_modified_unix_time = get_last_modified_time(file_content)
+        if last_modified_unix_time + DD_S3_CACHE_LOCK_TTL_SECONDS >= time():
+            return False
+    except ClientError:
+        pass
```

Comment:
Let's add a debugging log with the error and traceback printed.

Reply:
✅